File: Backend/main-backend-app/room-service/src/main/java/rs/ac/uns/ftn/xml/team17/roomservice/dto/room/RoomFull.java
package rs.ac.uns.ftn.xml.team17.roomservice.dto.room;

import java.util.List;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import rs.ac.uns.ftn.xml.team17.roomservice.dto.hotel.HotelPreview;
import rs.ac.uns.ftn.xml.team17.roomservice.model.additionalService.AdditionalService;
import rs.ac.uns.ftn.xml.team17.roomservice.model.address.Address;
import rs.ac.uns.ftn.xml.team17.roomservice.model.roomCategory.RoomCategory;
import rs.ac.uns.ftn.xml.team17.roomservice.model.roomType.RoomType;

@NoArgsConstructor
@Getter
@Setter
public class RoomFull {
	private Integer id;
	private HotelPreview hotel; 
	private Address address;
	private RoomType type;
	private RoomCategory category;
	private List<AdditionalService> additionalServices;
	private Integer cancelationDays;
	private Integer numberOfPeople;
	private Integer roomNumber;
	private Integer floorNumber;
	private String description;
	private Double totalRating;
	private Integer numberOfRatings;
}

File: Frontend/client-front/src/app/model/room/roomFull.ts
import { HotelPreview } from '../hotel/hotelPreview';
import { Address } from '../address';
import { RoomType } from './roomType';
import { RoomCategory } from './roomCategory';
import { AdditionalService } from './additionalService';

export class RoomFull {
    id: number;
    hotel: HotelPreview = new HotelPreview();
    address: Address = new Address();
    type: RoomType = new RoomType();
    category: RoomCategory = new RoomCategory();
    additionalServices: AdditionalService[] = [];
    cancelationDays: number;
    numberOfPeople: number;
    roomNumber: number;
    floorNumber: number;
    description: string;
    defaultPrice: number; // double
    totalRating: number; // double
    numberOfRatings: number;
}
